weave_query/weave_query/io_service.py

```diff
@@ -217,6 +218,7 @@ def __init__(
         self.register_handler_fn("ensure_file", self.handle_ensure_file)
         self.register_handler_fn("direct_url", self.handle_direct_url)
         self.register_handler_fn("sleep", self.handle_sleep)
+        self.register_handler_fn("query_traces", self.handle_query_traces)
```

Comment:
while it was the intention for all IO to go through this service, in practice not all of the application actually abides by that, for example the graphql client executes queries directly from request handlers in the server. right now the IO service is used mainly for handling files that are present either the local fs or a remote fs. HTTP requests for REST apis generally are made from the server directly, and dont go through the IO service.

was there a specific motivation for putting this in the io service? as a first step, it may be simpler / easier to test to call out to the trace server directly with `requests` in the handler code

Reply:
thank you for the background, Danny, super helpful. This is my first time working with python async code (I didn't know python had async / await until a few weeks ago), so I thought I would try to follow what's being done for concurrency reasons. 

let me tweak these set of changes so the request doesn't go through the IO service and follow your suggestion 👍